common/src/main/java/com/wynntils/features/SilencerFeature.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features;
+
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.config.Category;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigCategory;
+import com.wynntils.core.config.RegisterConfig;
+import com.wynntils.core.features.Feature;
+import com.wynntils.core.features.properties.RegisterKeyBind;
+import com.wynntils.core.keybinds.KeyBind;
+import com.wynntils.core.mod.event.WynncraftConnectionEvent;
+import com.wynntils.core.storage.Storage;
+import com.wynntils.mc.event.TitleScreenInitEvent;
+import com.wynntils.utils.mc.McUtils;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.sounds.SoundSource;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@ConfigCategory(Category.UNCATEGORIZED)
+public class SilencerFeature extends Feature {
+    @RegisterConfig
+    public final Config<Double> silencerVolume = new Config<>(0.01);
+
+    @RegisterKeyBind
+    private final KeyBind silencerKeyBind =
+            new KeyBind("Toggle Silencer", GLFW.GLFW_KEY_UNKNOWN, true, this::toggleSilencer);
+
+    private final Storage<Double> lastVolume = new Storage<>(1.0);
+
+    private boolean isSilencerEnabled = false;
+
+    @SubscribeEvent
+    public void onDisconnect(WynncraftConnectionEvent.Disconnected event) {
+        if (isSilencerEnabled) {
+            resetVolume();
+            isSilencerEnabled = false;
+        }
+    }
+
+    @SubscribeEvent
+    public void onTitleScreenInit(TitleScreenInitEvent.Pre event) {
+        resetVolume();
```

Comment:
... otherwise you can easily get cases like this, where the volume is actually reset, but we do not track this so we still think the silencer is enabled.

Reply:
Also, why is this needed?

Reply:
Indeed, the `onDisconnect` function does handle a similar task, but as @kristofbolyai pointed out, we can't guarantee that it always successfully resets the volume, which might lead to unreliable behavior.
Hence, we reset the volume at the start of the game, specifically just before the `TitleScreen` loads.
Additionally we keep the `onDisconnect` to handle scenarios where the user disconnects from the server but doesn't return to the `TitleScreen`.

Regarding why we don't set `isSilencerEnabled` in `onTitleScreenInit`, the Boolean is defaulted to false on launch, but the volume is not reset. Furthermore, accessing the TitleScreen is only possible after disconnecting from the server, a prerequisite to set the Boolean to true.

Reply:
So is this an attempt to restore the volume at startup in case of a crash? I think you should better do that in the constructor. That will fix the volume for the entire minecraft game as soon as the mod loads. This event will be sent only when our event framework is up and running. And it will be resent (I think) whenever you get back to the title screen from a game.

Reply:
> And it will be reset (I think) whenever you get back to the title screen from a game.

Yes, you're correct. In the current design, the volume is reset to `lastVolume` every time you return to the Title Screen, which indeed is an oversight. To address this, I suggest introducing a new local variable `currentVolume`. We would then perform a check to see if `currentVolume` equals `silencerVolume`. Based on this check, we either proceed with resetting to `lastVolume` or do nothing. This check ensures that we only reset the volume when necessary, which should resolve the issue.

> So is this an attempt to restore the volume at startup in case of a crash?

No, not exactly. The intention behind this design isn't specifically to restore the volume at startup in case of a crash. Instead, it's primarily to reapply the user's vanilla volume value.

Reply:
Ok, I think I understand this process a bit better now. There is a lot of cases, and I am not sure these are all correctly handled, or even what the correct way to handle them would be.

Imagine that you log onto Wynncraft and enable the silencer. Then you log out. I expect the normal vanilla volume to be restored. I believe this happens, and that the onConnect + onTitleScreen is basically a safeguard to make sure this really happens, regardless.

But what happens when you log back on to Wynncraft? Should the silencer still be on? If so, we need to change the volume upon connection. If not, we must make sure we do not believe the silencer is enabled. It is still not obvious to me that we have such a guarantee, but maybe I'm just being stupid. :-)

To help with making the class understandable, can you make things clearer by renaming `lastVolume` to `originalVolume`? And `resetVolume` to `restoreOriginalVolume`?

Also, I would like for you to write some documentation (like a block at the top of the class) of how this all comes together, summarizing what you have described here in the comments.